lib/agenda/database.js

```diff
@@ -26,7 +26,8 @@ module.exports = function(url, collection, options, cb) {
 
   collection = collection || 'agendaJobs';
   options = Object.assign({autoReconnect: true, reconnectTries: Number.MAX_SAFE_INTEGER, reconnectInterval: this._processEvery}, options);
-  MongoClient.connect(url, options, (error, db) => {
+  MongoClient.connect(url, options, (error, client) => {
+    const database = client.s.options.dbName || 'agenda';
```

Comment:
I'm not sure how much we can trust this.

Reply:
Could parse the connection string? https://github.com/mongolab/mongodb-uri-node

Reply:
I like the sentiment but the URL string has seen some changes (and might do so in the future, too). I'm not too happy to rely on an external package which hasn't been updated in years.

Tests should start failing if something changes with that object, so I'd be confident enough to keep it.

A viable alternative would be to require manually configuring it and now would be a great time to make that kind of breaking change.